cpp/daal/src/externals/service_rng_openrng.h

```diff
@@ -0,0 +1,393 @@
+/* file: service_rng_openrng.h.h */
+/*******************************************************************************
+* Copyright contributors to the oneDAL project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef __SERVICE_RNG_OPENRNG_H__
+#define __SERVICE_RNG_OPENRNG_H__
+
+#include "openrng.h"
+#include "src/externals/service_stat_rng_ref.h"
+#include "src/externals/service_rng_common.h"
+
+// RNGs
+#define __DAAL_BRNG_MT2203                    VSL_BRNG_MT2203
+#define __DAAL_BRNG_MT19937                   VSL_BRNG_MT19937
+#define __DAAL_BRNG_MCG59                     VSL_BRNG_MCG59
+#define __DAAL_RNG_METHOD_UNIFORM_STD         VSL_RNG_METHOD_UNIFORM_STD
+#define __DAAL_RNG_METHOD_UNIFORMBITS32_STD   0
+#define __DAAL_RNG_METHOD_BERNOULLI_ICDF      VSL_RNG_METHOD_BERNOULLI_ICDF
+#define __DAAL_RNG_METHOD_GAUSSIAN_BOXMULLER  VSL_RNG_METHOD_GAUSSIAN_BOXMULLER
+#define __DAAL_RNG_METHOD_GAUSSIAN_BOXMULLER2 VSL_RNG_METHOD_GAUSSIAN_BOXMULLER2
+#define __DAAL_RNG_METHOD_GAUSSIAN_ICDF       VSL_RNG_METHOD_GAUSSIAN_ICDF
+
+namespace daal
+{
+namespace internal
+{
+namespace ref
+{
+/* Uniform distribution generator functions */
+template <typename T, CpuType cpu>
+int uniformRNG(const size_t n, T * r, void * stream, const T a, const T b, const int method);
+
+template <CpuType cpu>
+int uniformRNG(const size_t cn, size_t * r, void * stream, const size_t a, const size_t b, const int method)
+{
+    size_t n    = cn;
+    int errcode = 0;
+
+    if (a >= b)
+    {
+        return -3;
+    }
+
+    if (sizeof(size_t) == sizeof(unsigned int))
+    {
+        size_t len = b - a;
+        int na     = -(len / 2 + len % 2);
+        int nb     = len / 2;
+        int nn     = (int)n;
+        int * rr   = (int *)r;
+        __DAAL_VSLFN_CALL_NR_WHILE(, viRngUniform, (method, stream, nn, rr, na, nb), errcode);
```

Comment:
why do you use DAAL macros in this file? Are you sure that it will be opened and work as expected? Can it be replaced with the direct call of openrng functions?

Reply:
It can be replaced with direct calls to openrng functions.  It works as expected using daal macros (from service_stat_rng_ref.h) but, I will change them to direct function calls for the sake of readability.